### ISSUE TITLE
metadata csv file

### ISSUE THREAD

**Issue by LiuBodan** [Novartis/UMM-Discovery#1]
Hi,

could you please provide the metadata csv file or inform me how to generate it?

Many thanks

**Comment by worldteacher-global** [comment]
Hi LiuBodan,

 I am also working on using this architecture for a project I am working on. I've found that the metadata used in the notebook can be created from 2 csv files located on the Broad Institute website (https://bbbc.broadinstitute.org/BBBC021) towards the bottom of the page (BBBC021_v1_image.csv, BBBC021_v1_moa.csv) you will have to remap the column names as well. I also had to modify the training data to be the appropriate dimensions (within the notebook) to run the model (please let me know if you have to do the same). After making all these changes the notebook runs and makes predictions and cluster assignments. 
 
 One issue I am now facing is noticing that the loss is not changing over 50 epochs which suggests to that the model is not learning. I will adjust my learning rate but I am curious if anyone has suggestions as to why this is..
 
 Thank you for any advice,
 Kofi

**Comment by LiuBodan** [comment]
@worldteacher-global 
Hi Kofi,
May I see what your csv file look like? Below is my version of the csv:
[UMM_meta.csv](https://github.com/Novartis/UMM-Discovery/files/12811736/UMM_meta.csv)

For the training, I modified two parts:
1. I changed the pesudolabel col to read the MoA (what is your pesudolabel?)
2. I have a mismatch with the feature and embeded col, where the feature has dimension 128 rather than 64, still trying to figure that out.

I saw you have forked the repo, would you mind to update your fork with your csv files and changes in notebook? I will look throught that.
Many thanks,
Bodan

**Comment by worldteacher-global** [comment]
@LiuBodan 
Hi Bodan,

Here is my version of the csv :
[dataset(1).csv](https://github.com/Novartis/UMM-Discovery/files/12833196/dataset.1.csv)

The repo I forked is not up-to-date with what I am currently using which is on an Amazon Workspace. If you'd like to share your workspace with me I don't mind working with you to help you in your effort.

Best,
Kofi

**Comment by LiuBodan** [comment]
@worldteacher-global 
https://github.com/LiuBodan/UMM-Discovery
Here is my repo, I did not modify much of the code.

Why your csv file has compound and MoA as undefined? Will my csv file help you in any way?

It's probably easiser if you could add my discord: rubber#4698, or email me at bodan.liu@anu.edu.au and let's work it out together.

Best,
Bodan

**Comment by worldteacher-global** [comment]
@LiuBodan 
The csv file was created by merging the moa.csv and the image.csv together. The image dataset has compounds with unkonwn MoAs, so in return the merged  data returns NaN values to that row in the moa.csv (moa_compound column "left join"). I simply changed the value from empty to UNKNOWN for consistency. The UNKNOWN you see in the compound column "right join" was already in the image.csv dataset (Week 8).

**Comment by LiuBodan** [comment]
@worldteacher-global 
Thanks for the explaination, may I know how did you cope with the modifying the training data to be the appropriate dimensions?
Would be very valuable, if you could update your repo with your cloud service.

**Comment by worldteacher-global** [comment]
![image](https://github.com/Novartis/UMM-Discovery/assets/82952907/dbc70c44-7e08-4a13-96a1-68929e51361d)
![image](https://github.com/Novartis/UMM-Discovery/assets/82952907/19075fc6-8ac5-41a6-a6ac-f3ec0e594e32)

**Comment by LiuBodan** [comment]
@worldteacher-global 
hi, thanks for sharing your code, however, I'm not sure how to use them without further context. Would be nice if you could update your jupyter notebook on Github so that I can have a look.

With modifying my csv file, I'm able to run the notebook without modifying its code and got following outputs
![image](https://github.com/Novartis/UMM-Discovery/assets/123052690/dd9bdca5-3907-4d1b-861a-81ae2fe33301)
However, my kernel will die after this step for unknown reason.

I have shared my new csv file below, please give it a try and tell me what you have.

[UMM_meta2.csv](https://github.com/Novartis/UMM-Discovery/files/13067759/UMM_meta2.csv)


For your csv file, I think there are 2 issues:
1. pesudoclass is undefined for all the samples
2. I think the paper's model is only trained on the 3800+ annotated images, but your csv will read all the images on BBBC021

Please let me know what you think,
Bodan

**Comment by worldteacher-global** [comment]
@LiuBodan  did you have any luck with your problem?

**Comment by worldteacher-global** [comment]
Still a work in progress, yourself?

**Comment by LiuBodan** [comment]
@worldteacher-global not really been doing some other project.....
